File: MyBatis/MyBatis/BMSqliteDemo/XX/SqlSessionFactoryBuilder.m
//
//  SqlSessionFactoryBuilder.m
//  ProxyDemo
//
//  Created by liuweizhen on 2019/11/3.
//  Copyright © 2019 liuxing8807@126.com. All rights reserved.
//

#import "SqlSessionFactoryBuilder.h"
#import "Configuration.h"
#import "XMLConfigBuilder.h"
#import "DefaultSqlSessionFactory.h"

@implementation SqlSessionFactoryBuilder

- (id<SqlSessionFactory>)build:(NSString *)configPath {
    Configuration *config = [XMLConfigBuilder loadConfiguration:configPath];
    return [[DefaultSqlSessionFactory alloc] initWithConfig:config];
}

@end

File: _book/MyBatis/demos/CustomMyBatis/src/main/java/com/daliu/mybatis/sqlsession/SqlSessionFactoryBuilder.java
package com.daliu.mybatis.sqlsession;

import com.daliu.mybatis.cfg.Configuration;
import com.daliu.mybatis.sqlsession.defaults.DefaultSqlSessionFactory;
import com.daliu.mybatis.utils.XMLConfigBuilder;

import java.io.InputStream;

/**
 * 用于创建SqlSessionFactory对象
 */
public class SqlSessionFactoryBuilder {
    /**
     * 根据配置文件(SqlMapConfig.xml)创建SqlSessionFactory工厂
     * @param config 配置文件流
     * @return
     */
    public SqlSessionFactory build(InputStream config) {
        Configuration cfg = XMLConfigBuilder.loadConfiguration(config);
        return new DefaultSqlSessionFactory(cfg);
    }
}
